src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -512,9 +512,20 @@ private void OnAbsoluteFormTarget(TargetOffsetPathLength targetPath, Span<byte>
             previousValue == null || previousValue.Length != target.Length ||
             !StringUtilities.BytesOrdinalEqualsStringAndAscii(previousValue, target))
         {
-            // The previous string does not match what the bytes would convert to,
-            // so we will need to generate a new string.
-            RawTarget = _parsedRawTarget = target.GetAsciiStringNonNullCharacters();
+            try
+            {
+                // The previous string does not match what the bytes would convert to,
+                // so we will need to generate a new string.
+                RawTarget = _parsedRawTarget = target.GetAsciiStringNonNullCharacters();
+            }
+            catch (InvalidOperationException)
+            {
+                // GetAsciiStringNonNullCharacters throws an InvalidOperationException if there are
+                // invalid characters in the string. This is hard to understand/diagnose, so let's
+                // catch it and instead throw a more meaningful error. This matches the behavior in
+                // the origin-form case.
+                ThrowRequestTargetRejected(target);
```

Comment:
Can the IOE exception be captured and flow as an inner exception?

Reply:
@Tratcher Looking at the IOE I don't think there's anything there that would really be useful. This approach is also consistent with how we surface the rejected target errors in other places.